File: AMoAdiOSSdkDemo(Objective-C)/AMoAdiOSSdkDemo(Objective-C)/DisplayViewController.m
//
//  DisplayViewController.m
//  AMoAdiOSSdkDemo(Object-C)
//
//  Created by AMoAd on 2016/09/02.
//  Copyright © 2016年 AMoAd. All rights reserved.
//

#import "DisplayViewController.h"
#import "AMoAdView.h"

@interface DisplayViewController () <AMoAdViewDelegate>
@property AMoAdView *adView;
@end

@implementation DisplayViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self initAd];
}

- (void) initAd {
    self.adView = [[AMoAdView alloc ]init];
    self.adView.delegate = self;
    self.adView.horizontalAlign = AMoAdHorizontalAlignCenter;
    self.adView.verticalAlign = AMoAdVerticalAlignBottom;
    self.adView.rotateTransition = AMoAdRotateTransitionFlipFromLeft;
    self.adView.clickTransition = AMoAdClickTransitionJump;
    self.adView.adjustMode = AMoAdAdjustModeResponsive;
    self.adView.sid = self.sid;
    [self.view addSubview:self.adView];
    
}
- (void) showAd {
    self.adView.hidden = false;
}
- (void) hideAd {
    self.adView.hidden = true;
}
- (IBAction)onHideBtnClicked:(id)sender {
    [self hideAd];
}
- (IBAction)onShowBtnClicked:(id)sender {
    [self showAd];
}

- (void)AMoAdViewDidReceiveAd:(AMoAdView *)amoadView {
    // 正常に広告を受信した
}

- (void)AMoAdViewDidFailToReceiveAd:(AMoAdView *)amoadView error:(NSError *)error {
    // 広告の取得に失敗した
}
- (void)AMoAdViewDidReceiveEmptyAd:(AMoAdView *)amoadView {
    // 空の広告を受信した
}

@end

File: AMoAdiOSSdkDemo/AMoAdiOSSdkDemo/DisplayViewController.swift
//
//  DisplayViewController.swift
//  AMoAdiOSSdkDemo
//
//  Created by AMoAd on 2016/08/15.
//  Copyright © 2016年 AMoAd. All rights reserved.
//

import UIKit

class DisplayViewController: UIViewController, AMoAdViewDelegate {
    var sid:String!
    var adView:AMoAdView!
    override func viewDidLoad() {
        super.viewDidLoad()
        initAd()
    }
    
    func initAd() {       
        adView = AMoAdView()
        adView.delegate = self
        adView.horizontalAlign = .center
        adView.verticalAlign = .bottom
        adView.rotateTransition = .flipFromLeft
        adView.clickTransition = .jump
        adView.adjustMode = .responsive
        adView.sid = sid
        self.view.addSubview(adView)
    }
    
    func showAd() {
        adView.show()
    }
    
    func hideAd() {
        adView.hide()
    }
    
    @IBAction func onHideBtnClicked(_ sender: AnyObject) {
        hideAd()
    }
    @IBAction func onShowBtnClicked(_ sender: AnyObject) {
        showAd()
    }

    func aMoAdViewDidReceiveAd(_ amoadView: AMoAdView!) {
        // 正常に広告を受信した
        print("AMoAdViewDidReceiveAd()")
    }
    
    func aMoAdViewDidFail(toReceiveAd amoadView: AMoAdView!, error: Error!) {
        // 広告の取得に失敗した
        print("AMoAdViewDidFailToReceiveAd()")
    }
    
    func aMoAdViewDidReceiveEmptyAd(_ amoadView: AMoAdView!) {
        // 空の広告を受信した
        print("AMoAdViewDidReceiveEmptyAd()")
    }
}
